### ISSUE TITLE
Pre-training error on my own dataset

### ISSUE THREAD

**Issue by alize16** [cantinilab/scPRINT#33]
I am trying to train scPrint on my own dataset to predict the disease based on the cells per patient. I believe I only need disease_ontology_term_id and organism_ontology_term_id for re-training. However, I keep getting the following error:

<img width="996" alt="Image" src="https://github.com/user-attachments/assets/70153f7e-554d-4803-ba1e-8e12aff98cfa" />

**My pretrain_mydata.yml file looks like this:**

# general params
project: my_scprint_pretraining #project name for saving data and wandb
seed_everything: 42
ckpt_path: null #we don't have a checkpoint weights as we train from scratch
set_float32_matmul_precision: True
wandblog: all #we use wandb here
log_freq: 200
log_graph: True

trainer: #training level params
  precision: 16-mixed #we use mixed precision 16bit for training
  gradient_clip_val: 100 #needed
  log_every_n_steps: 100
  logger: #we can add multiple loggers (see below)
    - class_path: lightning.pytorch.loggers.WandbLogger
  callbacks: #you can create your own callback and add it here or use lightning's callbacks
    - class_path: lightning.pytorch.callbacks.StochasticWeightAveraging
      init_args:
        swa_lrs: 0.03

model: # model params
  dropout: 0.1
  transformer: flash #flashattention is used
  mvc_decoder: inner product
  residual_in_fp32: True

data:
    collection_name: preprocessed_mydata
    organisms:
      - NCBITaxon:9606
    gene_position_tolerance: 10000
    gene_embeddings: ./data/main/gene_embeddings.parquet
    how: random expr
    max_len: 2200
    weight_scaler: 50
    hierarchical_clss:
      - disease_ontology_term_id
    # - cell_type_ontology_term_id
    # - disease_ontology_term_id
    # - assay_ontology_term_id
    # - self_reported_ethnicity_ontology_term_id
    clss_to_weight:
    - disease_ontology_term_id
    - organism_ontology_term_id
    # - cell_type_ontology_term_id
    # - disease_ontology_term_id
    # - assay_ontology_term_id
    # - self_reported_ethnicity_ontology_term_id
    # - sex_ontology_term_id

    clss_to_predict:
      - disease_ontology_term_id
      - organism_ontology_term_id
    # - cell_type_ontology_term_id
    # - disease_ontology_term_id
    # - assay_ontology_term_id
    # - self_reported_ethnicity_ontology_term_id
    # - sex_ontology_term_id
    # - organism_ontology_term_id 

**I am giving the following command in terminal to train on my dataset:**
scprint fit \
  --config config/base.yml \
  --config config/pretrain_mydata.yml \
  --scprint_training.name=my_disease_model \
  --trainer.strategy=auto

**Comment by jkobject** [comment]
Ok I think I would need the full stack trace too but a guess:
in scdataloader there is this line option "assay_to_drop" which default is actually a set of assay I think should always be dropped.
However this triggers this line:
```python
          if len(self.assays_to_drop) > 0:
                badloc = np.isin(
                    self.dataset.mapped_dataset.get_merged_labels(
                        "assay_ontology_term_id"
                    ),
                    self.assays_to_drop,
                )
                idx_full = np.arange(len(labels))[~badloc]
```

which can cause an issue. just set this parameter to an empty string in you parameter's file: something like this:
- data.assays_to_drop: []

**Comment by jkobject** [comment]
Also since scPRINT is a foundation model, the idea is that it is always better not to train from scratch and do fine-tuning instead from a pre-trained model.

**Comment by jkobject** [comment]
Have you tried the parameter the solution?
`data.assays_to_drop: []`

Yes, but this is called fine-tuning what you want to do. you start from a model that has already learnt about cells and just train it a bit more on this task so that it gets better. From scratch, the model is too big to train on your dataset.

**Comment by jkobject** [comment]
Hello @alize16 ,

to answer your issue with loading models, you sent by email:
I can tell you that with the scprint latest version on main I was able to load both large.ckpt and v2-medium.ckpt.

However as you can see in all bench_*.ipynb notebooks now I use a couple lines of code to do the loading and make sure that both model versions can load:

```python
m = torch.load(model_checkpoint_file)
if "prenorm" in m['hyper_parameters']:
    m['hyper_parameters'].pop("prenorm")
    torch.save(m, model_checkpoint_file)
if "label_counts" in m['hyper_parameters']:
    model = scPrint.load_from_checkpoint(model_checkpoint_file, precpt_gene_emb=None, classes=m['hyper_parameters']['label_counts'])
else:
    model = scPrint.load_from_checkpoint(model_checkpoint_file, precpt_gene_emb=None)
```

Can you confirm that the checkpoints still don't load on your end with this?

**Comment by jkobject** [comment]
Hey, I will be closing this without any new updates :)
